packages/kbn-test/src/page_load_metrics/navigation.ts

```diff
@@ -0,0 +1,159 @@
+  await page.goto(loginUrl, {
+    waitUntil: 'networkidle0',
+  });
+  await page.type('[data-test-subj="loginUsername"]', options.appConfig.username);
+  await page.type('[data-test-subj="loginPassword"]', options.appConfig.password);
+  await page.click('[data-test-subj="loginSubmit"]');
+  await page.waitForNavigation({ waitUntil: 'networkidle0' });
+  await page.close();
+}
+
+export async function navigateToApps(log: ToolingLog, options: NavigationOptions) {
+  const browser = await puppeteer.launch({ headless: options.headless, args: ['--no-sandbox'] });
+  const devToolsResponses: NavigationResults = new Map();
+  const apps = [
+    { path: '/app/discover', locator: '[data-test-subj="discover-sidebar"]' },
+    { path: '/app/home', locator: '[data-test-subj="homeApp"]' },
+    { path: '/app/canvas', locator: '[data-test-subj="create-workpad-button"]' },
+    { path: '/app/maps', locator: '[data-test-subj="mapContainer"]' },
+    { path: '/app/apm', locator: '[data-test-subj="apmMainContainer"]' },
+  ];
+
+  await loginToKibana(log, browser, options);
+
+  await Promise.all(
+    apps.map(async (app) => {
+      const page = await browser.newPage();
+      page.setCacheEnabled(false);
+      page.setDefaultNavigationTimeout(0);
+      const frameResponses = new Map<string, FrameResponse>();
+      devToolsResponses.set(app.path, frameResponses);
+
+      const client = await page.target().createCDPSession();
+      await client.send('Network.enable');
+
+      function getRequestData(requestId: string) {
+        if (!frameResponses.has(requestId)) {
+          frameResponses.set(requestId, { url: '', dataLength: 0 });
+        }
+
+        return frameResponses.get(requestId)!;
+      }
+
+      client.on('Network.responseReceived', (event: ResponseReceivedEvent) => {
+        getRequestData(event.requestId).url = event.response.url;
+      });
+
+      client.on('Network.dataReceived', (event: DataReceivedEvent) => {
+        getRequestData(event.requestId).dataLength += event.dataLength;
+      });
+
+      const url = createUrl(app.path, options.appConfig.url);
+      log.debug(`goto ${url}`);
+      await page.goto(url, {
+        waitUntil: 'networkidle0',
+      });
+
+      let readyAttempt = 0;
+      let selectorFound = false;
+      while (!selectorFound) {
+        readyAttempt += 1;
+        try {
+          await page.waitForSelector(app.locator, { timeout: 5000 });
+          selectorFound = true;
+        } catch (error) {
+          log.error(
+            `Page '${app.path}' was not loaded properly, unable to find '${
+              app.locator
+            }', url: ${page.url()}`
+          );
+
+          if (readyAttempt < 6) {
+            continue;
+          }
+
+          await page.bringToFront();
+          Fs.mkdirSync(failureDir, { recursive: true });
+          await page.screenshot({
+            path: resolve(failureDir, `${app.path.replace('/', '_')}_navigation.png`),
```

Comment:
Doesn't look like these screenshots are being uploaded to GCP. I'm not confident in this, but it might be caused by the fact that there are multiple `/` characters in `app.path` and this is only replacing the first one. Try `app.path.split('/').join('_')`.

Additionally, we should probably write the failure screenshots to a directory that's configured by the user rather than hard coding `x-pack/test/...` at the top of this file.

Reply:
The default for the CLI should probably be just to write the screenshots to the CWD